### PR TITLE
Steb/refactor

### PR DESCRIPTION
**Pull request by Stebalien**
Draft PR to track pending changes. I'm making individual PRs for each commit in this branch.

Changes:

* Added documentation. Including documentation of method runtime.
* Optimize internals such that unsetting a bit doesn't turn all operations into `O(bits)` time.
* Make the bitfield receiver a pointer (there's no point in copying this stuff around).
* Optimize IsEmpty to make it `O(1)`.
* Benchmark.

### CODE REVIEW

**Comment by Stebalien** [comment]
Benchmark comparison: 

```
benchmark                               old ns/op     new ns/op     delta
name                        old time/op  new time/op  delta
Count/0/basic-4              182ns ±11%   172ns ± 8%   -5.60%  (p=0.045 n=10+8)
Count/1/basic-4              177ns ± 1%   181ns ±13%     ~     (p=0.878 n=8+10)
Count/1/modified-4           935ns ± 2%  1021ns ± 2%   +9.26%  (p=0.000 n=9+9)
Count/10/basic-4             951ns ±17%   899ns ± 5%   -5.46%  (p=0.007 n=9+9)
Count/10/modified-4         1.78µs ± 4%  1.61µs ± 3%   -9.83%  (p=0.000 n=8+8)
Count/1000/basic-4           107µs ± 3%   102µs ± 3%   -4.43%  (p=0.000 n=10+10)
Count/1000/modified-4        177µs ± 4%   112µs ± 4%  -36.62%  (p=0.000 n=9+10)
Count/1000000/basic-4        190ms ±11%   173ms ± 7%   -8.50%  (p=0.001 n=9+9)
Count/1000000/modified-4     265ms ± 4%   222ms ±37%     ~     (p=0.143 n=10+10)
IsEmpty/0/basic-4            222ns ±59%   182ns ± 1%     ~     (p=0.174 n=10+9)
IsEmpty/1/basic-4            209ns ±16%   182ns ± 2%  -12.56%  (p=0.000 n=10+10)
IsEmpty/1/modified-4         942ns ± 1%  1087ns ± 1%  +15.43%  (p=0.000 n=9+9)
IsEmpty/10/basic-4           970ns ±12%   801ns ± 2%  -17.42%  (p=0.000 n=10+9)
IsEmpty/10/modified-4       1.78µs ± 4%  1.61µs ± 1%   -9.60%  (p=0.000 n=9+9)
IsEmpty/1000/basic-4         106µs ± 2%    83µs ± 2%  -21.02%  (p=0.000 n=7+10)
IsEmpty/1000/modified-4      171µs ± 1%   112µs ± 4%  -34.16%  (p=0.000 n=9+10)
IsEmpty/1000000/basic-4      175ms ± 4%   165ms ± 9%   -5.76%  (p=0.001 n=10+9)
IsEmpty/1000000/modified-4   267ms ± 6%   189ms ±14%  -29.11%  (p=0.000 n=8+10)

```

Name format: "MethodBeingTested/NumberOfElements/subtest" where "subtest" is "basic" for unmodified and "modified" for a bitfield with bits set/unset with `Unset` and Set`.

**Comment by Stebalien** [comment]
There's a bit of added complexity for tiny (size 1) bitfields, but that shouldn't matter in practice.

File: bitfield.go [hunk 1]
```diff
@@ -73,6 +80,18 @@ func NewFromIter(r rlepluslazy.RunIterator) (*BitField, error) {
...
+// MergeBitFields returns the union of the two BitFields.
+//
+// For example, given two BitFields:
+//
+//     0 1 1 0 1
+//     1 1 0 1 0
+//
+// MergeBitFields would return
+//
+//     1 1 1 1 0
```
Comment (Kubuxu) [inline bitfield.go:92]: Should be `1 1 1 1 1`

**Comment by Stebalien** [comment]
Last patch in #29.

**Pull request closed without merging**